RTE Macro Parser

aochmann opened the pull request:
<!--
Thank you for your pull request, #h5yr! But it's just a beginning. 
Please provide a details where it's required and review the requirements below.
-->
**What this PR does / why it's submitted / why we need it**:
Added support for parsing macro elements inside RTE. Code coverage by tests.

---

**Checklist**:
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] My PR has a descriptive title (not a vague title like `Needed this`)
- [x] All the commits follows [commitlint guidelines](https://github.com/conventional-changelog/commitlint#what-is-commitlint)
- [x] My PR targets the `develop` branch or appropriate `release` branch

---

aochmann committed:
feat: added rte macro parser with tests

aochmann committed:
feat: added generic and macro related tests

mzajkowski commented on Source/Our.Umbraco.Nexu.Parsers.Tests/Our.Umbraco.Nexu.Parsers.Tests.csproj:
```diff
@@ -302,6 +302,7 @@
     <Compile Include="GridEditorParsers\Core\MediaGridEditorParserTests.cs" />
     <Compile Include="GridEditorParsers\Core\RteGridEditorParserTests.cs" />
     <Compile Include="PropertyParsers\Community\ContentListParserTests.cs" />
+    <Compile Include="PropertyParsers\Community\RichTextMacrosParserTests.cs" />
```
Rename this class to singular "Macro" following the other namings there.

mzajkowski commented on Source/Our.Umbraco.Nexu.Parsers.Tests/PropertyParsers/Community/RichTextMacrosParserTests.cs:
```diff
@@ -0,0 +1,392 @@
+        private Mock<ICacheProvider> _cacheProviderMock;
+
+
+        #region Test data
```
ooooo regions <3

aochmann committed:
fix: renamed rte macro class and file names

To match project naming convention

mzajkowski reviewed (approved):
One of the best PRs and work items I've seen for a long time. Well structured PR, well placed within the existing code-base, covered with tests and proper naming conventions.

![giphy](https://user-images.githubusercontent.com/11617260/70528614-e7e27b00-1b4e-11ea-8a29-9ab342a6d416.gif)

aochmann committed:
feat: added rte parser for macros without comments

aochmann committed:
feat: changed getting property parser for same data type definition

This will allow to create dedicated custom property parser with dedicated responsibility
